exercises.py

```diff
@@ -1,28 +1,53 @@
-"""
-Recursion Problem Set: Coding Exercises
-"""
+"""Recursion Problem Set: Coding Exercises"""
 
 def reverse(s: str) -> str:
-    """
-    Reverse string using recursive approach
-    """
 
+    if len(s) == 0 or len(s) == 1:
+        return s
+    else:
+        return reverse(s[1:]) + s[0]    
+        
 def fib_rec(n: int) -> int:
-    """
-    Recursive fibonacci
-    """
+    """Recursive fibonacci"""
+    if n == 1 or n == 0:
+        return 1
+    
+    return fib_rec(n - 1) + fib_rec(n-2)
 
 def fib_dyn(n: int) -> int:
-    """
-    Fibonacci using dynamic programming (w/ a cache)
-    """
+    """Fibonacci using dynamic programming (w/ a cache)"""
+    cache = {}
+    
+    if n == 1 or n == 0:
+        return 1
+    
+    if n in cache:
+        return cache[n]
+    else:
+        cache[n] = fib_dyn(n - 1) + fib_dyn(n - 2)
+        return cache[n]
 
 def fib_iter(n: int) -> int:
-    """
-    Iterative approach to solving the fibonacci problem
-    """
+    """Iterative approach to solving the fibonacci problem"""
+    
+    last_number = 1
+    two_before = 1
+    
+    if n == 0 or n == 1:
+        return 1
```

Comment:
Return `n` here too, see [Fibonacci number](https://en.wikipedia.org/wiki/Fibonacci_number). Rest of block looks good! Great variable names!